docs/machine-learning/how-does-mldotnet-work.md

```diff
@@ -0,0 +1,262 @@
+```
+
+In the snippet, `Concatenate` and `Sdca` are both extension functions. They each create an <xref:Microsoft.ML.IEstimator`1> object that is appended to the pipeline. At this point, the objects are created only. No execution has happened.
+
+### Fit the model
+
+Once the objects in the pipeline have been created, data can be used to train or fit the model.
+
+```csharp
+    var model = pipeline.Fit(trainingData);
+```
+
+Calling `Fit()` uses the input training data to estimate the parameters of the model. Remember in our linear regression model above, we had two model parameters: **bias** and **weight**. After the `Fit()` call, the values of the parameters are known. Most models will have many more parameters than this.
+
+You can learn more about model training in [How to train your model](./how-to-guides/train-model.md)
+
+The resulting model object implements the <xref:Microsoft.ML.ITransformer> interface. That is, the model transforms transforms input data into predictions. 
+
+### Transform data
+
+You can transform input data into predictions in bulk, or one input at a time. In our example, we did both: in bulk for the purpose of evaluating the model, and one at a time to make a new prediction. Let's look at making single predictions.
+
+```csharp
+    var size = new HouseData() { Size = 2.5F };
+    var predEngine = model.CreatePredictionEngine<HouseData, Prediction>(mlContext);
+    var price = predEngine.Predict(size);
+```
+ 
+The `CreatePredictionEngine()` method takes an input class and an output class. The field names and/or code attributes determine the names of the data columns used during model training and prediction. You can read about  [How to make a single prediction](./how-to-guides/single-predict-model-ml-net.md) in the How-to section.
+
+### Data models and schema
+
+At the core of an ML.NET machine learning pipeline are [DataView](xref:Microsoft.ML.IDataView) objects.
+
+Each transformation in the pipeline has an input schema (data names, types, and sizes that the transform expects to see on its input); and an output schema (data names, types, and sizes that the transform produces after the transformation). 
+
+If the output schema from one transform in the pipeline does not match the input schema of the next transform, ML.NET will throw an exception.
+
+A data view object has columns and rows. Each column has a name and a type and a length. For example: the input columns in the house price example are **Size** and **Price**. They are both type <xref:System.Single> and they are scalar quantities rather than vector ones.
+
+   ![ML.NET Data View example with house price prediction data](./media/ml-net-dataview.png)
+
+All ML.NET algorithms look for an input column that is a vector. By default this vector column is called **Features**. This is why we concatenated the **Size** column into a new column called **Features** in our house price example.
```

Comment:
"This is why the **Size** column was concatenated into a new column called **Features** in the house price example."